src/utils/messaging-outbox/src/services/outbox_transactional_processor.rs

```diff
@@ -509,3 +530,39 @@ impl ProducerRelayJob {
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug)]
+
+pub struct OutboxTransactionalProcessorMetrics {
+    pub failed_consumers: prometheus::IntGaugeVec,
```

Comment:
For my understanding, why using _gauge_ and not a _counter_?
Looking at definitions in Prometheus docs:

>A _counter_ is a cumulative metric that represents a single monotonically increasing counter whose value **can only increase or be reset to zero on restart**.

> A _gauge_ is a metric that represents a single numerical value that can arbitrarily go up and down.

Reply:
I added a few more metrics just to give them a good try:
- counter `outbox_messages_processed_total{producer,consumer}`
- gauge `outbox_messages_pending_total{producer,consumer}` - this one measures the lag behing latest ID and the last processed ID - also a great alert candidate